packages/e2e/lib/e2e.dart

```diff
@@ -128,9 +137,57 @@ class E2EWidgetsFlutterBinding extends LiveTestWidgetsFlutterBinding {
   /// The default value is `null`.
   Map<String, dynamic> reportData;
 
+  /// Tests will put the action requests from WebDriver to this pipe.
+  Completer<WebDriverAction> webDriverActionPipe = Completer<WebDriverAction>();
+
+  /// Updated when WebDriver completes the request by the test method.
+  ///
+  /// For example, a test method will ask for a screenshot by calling
+  /// `takeScreenshot`. When this screenshot is taken [driverActionComplete]
+  /// will complete.
+  Completer<bool> driverActionComplete = Completer<bool>();
+
+  final Mutex webDriverCommandLock = new Mutex();
+
+  /// Takes screenshot using WebDriver screenshot command.
+  ///
+  /// Only works on Web when tests are run via `flutter driver` command.
+  ///
+  /// See: https://www.w3.org/TR/webdriver/#screen-capture
+  Future<void> takeScreenshot(String screenshot_name) async {
+    if (!kIsWeb) {
+      throw StateError('Web driver commands are only avaliable on web');
```

Comment:
I think this should probably be an `UnimplementedError`. This doesn't have to be web driver specific right?

Reply:
done.